### PR TITLE
Feature/616

### LINKED ISSUE #616: [Feature]: Apply localization/translation to StatsScoringDialog
### Feature Summary

We should use i18n to translate the content of the `StatsScoringDialog`

### Detailed Description

We should apply the same i18n translations to `StatsScoringDialog` by putting the text into the translation files under the "global" block in a new "global.statsScoring" key. This entails pulling all the text from the `StatsScoringDialog` into the language files e.g. `en.json`, `es.json` `fr.json` and rendering the text with `i18n`'s `t()` function. 

See `LoginView.vue` and `HomeView.vue` for examples

### PR DESCRIPTION
**Pull request by arnaut08**
## 616

Relevant [issue number](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- Resolves #616

## Please check the following

- [x] Do the tests still pass? (see [Run the Tests](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#run-the-tests))
- [x] Is the code formatted properly? (see [Linting (Formatting)](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#linting-formatting))
- For New Features:
  - [ ] Have tests been added to cover any new features or fixes?
  - [ ] Has the documentation been updated accordingly?

### CODE REVIEW

File: src/components/StatsScoringDialog.vue [hunk 1]
```diff
@@ -17,73 +17,66 @@
           variant="text"
           data-cy="ranked-info-button"
         >
-          <span v-if="showButtonText">How are ranks determined?</span>
+          <span v-if="showButtonText">{{ t('game.dialogs.statsScoringDialog.ranksDetermination') }}</span>
```
Comment (itsalaidbacklife) [inline src/components/StatsScoringDialog.vue:20]: @arnaut08 let's rename this string to `rankDetermination`
```suggestion
          <span v-if="showButtonText">{{ t('game.dialogs.statsScoringDialog.rankDetermination') }}</span>
```

File: src/components/StatsScoringDialog.vue [hunk 2]
```diff
@@ -17,73 +17,66 @@
...
-            2 Points
+            2 {{ t('game.dialogs.statsScoringDialog.points.displayName') }}
           </v-chip>
-          Each other player who won at least one match gets 2 points for the week
+          {{ t('game.dialogs.statsScoringDialog.points.description.2Points') }}
         </v-list-item>
         <v-list-item>
           <v-chip variant="outlined" class="mr-2 mb-1" color="#000">
-            1 Point
+            1 {{ t('game.dialogs.statsScoringDialog.points.singular') }}
```
Comment (itsalaidbacklife) [inline src/components/StatsScoringDialog.vue:66]: We actually don't need a separate block for this, we can use i18n's pluralization system to specify the count like so 
```suggestion
            {{ t('game.dialogs.statsScoringDialog.points.displayName', 1) }}
```

Then we can correspondingly update all the other references to use this new definition and update the definition itself in the language files to specify the count and to use a different word for the singular version

File: src/components/StatsScoringDialog.vue [hunk 3]
```diff
@@ -17,73 +17,66 @@
...
-        Each Season is divided into 13 weeks. For each week, we count the number of best 2/3 ranked matches
-        each player wins against unique opponents (ignoring stalemates) and assign players points based on
-        their weekly standing.
+        {{ t('game.dialogs.statsScoringDialog.seasonDescription') }}
       </p>
       <v-list class="mt-4" bg-color="surface-2" base-color="surface-1">
         <v-list-item>
           <v-chip variant="elevated" class="mr-2 mb-1" :color="theme.firstPlace">
-            5 Points
+            5 {{ t('game.dialogs.statsScoringDialog.points.displayName') }}
```
Comment (itsalaidbacklife) [inline src/components/StatsScoringDialog.vue:42]: Let's refactor the points displayName to input a count and pluralize using the same key
```suggestion
            {{ t('game.dialogs.statsScoringDialog.points.displayName', 5) }}
```

File: src/components/StatsScoringDialog.vue [hunk 4]
```diff
@@ -17,73 +17,66 @@
...
           </v-chip>
-          The player with the most wins gets 5 points for the week
+          {{ t('game.dialogs.statsScoringDialog.points.description.5Points') }}
         </v-list-item>
         <v-list-item>
           <v-chip variant="elevated" class="mr-2 mb-1" :color="theme.secondPlace">
-            4 Points
+            4 {{ t('game.dialogs.statsScoringDialog.points.displayName') }}
```
Comment (itsalaidbacklife) [inline src/components/StatsScoringDialog.vue:48]:
```suggestion
            {{ t('game.dialogs.statsScoringDialog.points.displayName', 4) }}
```

File: src/components/StatsScoringDialog.vue [hunk 5]
```diff
@@ -17,73 +17,66 @@
...
           </v-chip>
-          The player with the 2nd most wins gets 4 points for the week
+          {{ t('game.dialogs.statsScoringDialog.points.description.4Points') }}
         </v-list-item>
         <v-list-item>
           <v-chip variant="elevated" class="mr-2 mb-1" :color="theme.thirdPlace">
-            3 Points
+            3 {{ t('game.dialogs.statsScoringDialog.points.displayName') }}
```
Comment (itsalaidbacklife) [inline src/components/StatsScoringDialog.vue:54]:
```suggestion
            {{ t('game.dialogs.statsScoringDialog.points.displayName', 3) }}
```

File: src/components/StatsScoringDialog.vue (hunk 2 above)
Comment (itsalaidbacklife) [reply src/components/StatsScoringDialog.vue:66]: You can see an example of this in ScoreGoalTooltip.vue in the `kingPoints` computed property which pluralizes the `game.score.kings` string

File: src/components/StatsScoringDialog.vue [hunk 6]
```diff
@@ -17,73 +17,66 @@
...
           </v-chip>
-          The player with the 3rd most wins gets 3 points for the week
+          {{ t('game.dialogs.statsScoringDialog.points.description.3Points') }}
         </v-list-item>
         <v-list-item>
           <v-chip variant="outlined" class="mr-2 mb-1" :color="theme.primary">
-            2 Points
+            2 {{ t('game.dialogs.statsScoringDialog.points.displayName') }}
```
Comment (itsalaidbacklife) [inline src/components/StatsScoringDialog.vue:60]:
```suggestion
            {{ t('game.dialogs.statsScoringDialog.points.displayName', 2) }}
```

File: src/translations/en.json [hunk 7]
```diff
@@ -92,6 +92,28 @@
...
+      "statsScoringDialog": {
+        "ranksDetermination": "How are ranks determined?",
+        "competitiveCuttle": {
+          "description": "Competitive Cuttle is divided into 4 seasons per year, one per suit: Clubs, Diamonds, Hearts, and Spades. At the end of each season, the top 8 players play a double elimination championship tournament and the",
+          "highlight": "champions are permanantly accoladed on the site."
+        },
+        "seasonDescription": "Each Season is divided into 13 weeks. For each week, we count the number of best 2/3 ranked matches each player wins against unique opponents (ignoring stalemates) and assign players points based on their weekly standing.",
+        "points": {
+          "displayName": "Points",
+          "singular": "Point",
```
Comment (itsalaidbacklife) [inline src/translations/en.json:105]: We can apply `i18n` pluralization to address both of these with a single key
```suggestion
          "displayName": "1 Point | {count} Points",
```

File: src/translations/es.json [hunk 8]
```diff
@@ -92,6 +92,28 @@
...
+      "statsScoringDialog": {
+        "ranksDetermination": "¿Cómo se determinan los rangos?",
+        "competitiveCuttle": {
+          "description": "Cuttle competitivo se divide en 4 temporadas por año, una por palo: Tréboles, Diamantes, Corazones y Espadas. Al final de cada temporada, los 8 mejores jugadores juegan un campeonato de doble eliminación y el",
+          "highlight": "Los campeones son permanentemente reconocidos en el sitio."
+        },
+        "seasonDescription": "Cada temporada se divide en 13 semanas. Para cada semana, contamos el número de partidos mejor clasificados de 2/3 que cada jugador gana contra oponentes únicos (ignorando los puntos muertos) y asignamos puntos a los jugadores según su clasificación semanal.",
+        "points": {
+          "displayName": "Puntos",
+          "singular": "Punto",
```
Comment (itsalaidbacklife) [inline src/translations/es.json:105]:
```suggestion
          "displayName": "1 Punto | {count} Puntos",
```

File: src/translations/fr.json [hunk 9]
```diff
@@ -92,6 +92,28 @@
...
+      "statsScoringDialog": {
+        "ranksDetermination": "Comment sont déterminés les classements?",
+        "competitiveCuttle": {
+          "description": "Competitive Cuttle est divisé en 4 saisons par an, une par couleur : trèfle, carreau, cœur et pique. A la fin de chaque saison, les 8 meilleurs joueurs disputent un tournoi de championnat à double élimination et le",
+          "highlight": "les champions sont récompensés en permanence sur le site."
+        },
+        "seasonDescription": "Chaque saison est divisée en 13 semaines. Pour chaque semaine, nous comptons le nombre de matchs classés aux 2/3 que chaque joueur remporte contre des adversaires uniques (en ignorant les impasses) et attribuons des points aux joueurs en fonction de leur classement hebdomadaire.",
+        "points": {
+          "displayName": "Points",
+          "singular": "Point",
```
Comment (itsalaidbacklife) [inline src/translations/fr.json:105]:
```suggestion
          "displayName": "1 Point | {count} Points",
```

File: src/translations/en.json [hunk 10]
```diff
@@ -92,6 +92,28 @@
         "title": "Select a card",
         "cannotMove": "Oops! It seems like you can't play. Select a Jack to scrap.",
         "resolve": "Resolve"
+      },
+      "statsScoringDialog": {
```
Comment (itsalaidbacklife) [inline src/translations/en.json:96]: This dialog isn't in the `GameView` so how about we move this whole block into the new `stats` block you've created in #727?

**Review by itsalaidbacklife — CHANGES_REQUESTED** [review]
Nice work. I'm requesting a revision of the pluralization and to move the dialog strings into the `stats` block that's created in #727

**Comment by itsalaidbacklife** [comment]
@arnaut08 are you still interested in completing this issue?

**Comment by arnaut08** [comment]
Apologies for the delayed response. Yes, I'll cover those comments and push my changes in a couple of days.

**Comment by arnaut08** [comment]
I'm drafting this and the other PR of mine as I might not have enough time to work on the changes for now. But I'll try to work on them as soon as possible.

**Comment by itsalaidbacklife** [comment]
Closing as stale

**Pull request closed without merging**